### ISSUE TITLE
Bot announces wrong item caught

### ISSUE THREAD

**Issue by WantedFreak** [MrKinau/FishingBot#75]
Sometimes the bot announces it caught an enchanted book for every item it catches for 5-10 minutes then goes back to normal.
When the problem happens it keeps going even after restarting it.

Command I use to start the bot:
`java -jar FishingBot-2.7.7.jar -config config.json -logsdir fish/`

OS is Linux Mint 19.3

Here is the log
```
x@x:~/Downloads/fish$ cat log0.log
[2020-09-16 21:11:27] [INFO   ] Loaded config from: /home/x/Downloads/config.json 
[2020-09-16 21:11:27] [INFO   ] Try to authenticate x@protonmail.com:*************** 
[2020-09-16 21:11:27] [INFO   ] Authentication successful! 
[2020-09-16 21:11:27] [INFO   ] Pinging x:x with protocol of MC-1.16.3 
[2020-09-16 21:11:27] [INFO   ] Received pong: Unknown, Version: 1.16.3 (753), online: 4 
[2020-09-16 21:11:27] [INFO   ] Module "FishingModule" enabled! 
[2020-09-16 21:11:27] [INFO   ] Module "HandshakeModule" enabled! 
[2020-09-16 21:11:27] [INFO   ] Module "LoginModule" enabled! 
[2020-09-16 21:11:27] [INFO   ] Module "ChatProxyModule" enabled! 
[2020-09-16 21:11:27] [INFO   ] Module "ChatCommandModule" enabled! 
[2020-09-16 21:11:27] [INFO   ] Module "ClientDefaultsModule" enabled! 
[2020-09-16 21:11:28] [INFO   ] Login successful! 
[2020-09-16 21:11:28] [INFO   ] Name: x 
[2020-09-16 21:11:28] [INFO   ] UUID: x
[2020-09-16 21:11:28] [INFO   ] [CHAT] x joined the game 
[2020-09-16 21:11:28] [INFO   ] Achieved level 48 
[2020-09-16 21:11:28] [INFO   ] [CHAT] You have no new mail. 
[2020-09-16 21:11:28] [INFO   ] [CHAT] <x> I am level 48 now 
[2020-09-16 21:11:28] [INFO   ] [CHAT] Welcome, x! 
[2020-09-16 21:11:28] [INFO   ] [CHAT] Type /help for a list of commands. 
[2020-09-16 21:11:28] [INFO   ] [CHAT] Type /list to see who else is online. 
[2020-09-16 21:11:28] [INFO   ] [CHAT] Players online: 5 - World time: 7:31 PM 
[2020-09-16 21:11:29] [INFO   ] Starting fishing! 
[2020-09-16 21:11:29] [INFO   ] [CHAT] <x> FishingBot v2.7.7 - Starting fishing 
[2020-09-16 21:11:44] [INFO   ] Caught "enchanted_book" 
[2020-09-16 21:11:44] [INFO   ] -> UNBREAKING III 
[2020-09-16 21:11:44] [INFO   ] -> FLAME I 
[2020-09-16 21:11:44] [INFO   ] [CHAT] <x> FishingBot v2.7.7 - Caught "enchanted_book" 
[2020-09-16 21:11:44] [INFO   ] [CHAT] <x> -> UNBREAKING III 
[2020-09-16 21:11:45] [INFO   ] [CHAT] <x> -> FLAME I 
[2020-09-16 21:11:54] [INFO   ] Caught "enchanted_book" 
[2020-09-16 21:11:54] [INFO   ] -> UNBREAKING III 
[2020-09-16 21:11:54] [INFO   ] -> FLAME I 
[2020-09-16 21:11:54] [INFO   ] [CHAT] <x> FishingBot v2.7.7 - Caught "enchanted_book" 
[2020-09-16 21:11:54] [INFO   ] [CHAT] <x> -> UNBREAKING III 
[2020-09-16 21:11:55] [INFO   ] [CHAT] <x> -> FLAME I 
[2020-09-16 21:12:11] [INFO   ] Caught "enchanted_book" 
[2020-09-16 21:12:11] [INFO   ] -> UNBREAKING III 
[2020-09-16 21:12:11] [INFO   ] -> FLAME I 
[2020-09-16 21:12:11] [INFO   ] [CHAT] <x> FishingBot v2.7.7 - Caught "enchanted_book" 
[2020-09-16 21:12:11] [INFO   ] [CHAT] <x> -> UNBREAKING III 
[2020-09-16 21:12:11] [INFO   ] [CHAT] <x> -> FLAME I 
[2020-09-16 21:12:18] [INFO   ] Caught "salmon" 
[2020-09-16 21:12:27] [INFO   ] Caught "cod" 
[2020-09-16 21:12:43] [INFO   ] Caught "salmon" 
[2020-09-16 21:12:56] [INFO   ] Caught "pufferfish" 
[2020-09-16 21:13:14] [INFO   ] Caught "salmon" 
[2020-09-16 21:13:21] [INFO   ] Caught "cod" 
[2020-09-16 21:13:26] [INFO   ] Caught "saddle" 
[2020-09-16 21:13:37] [INFO   ] Caught "cod" 
[2020-09-16 21:13:46] [INFO   ] Caught "cod" 
[2020-09-16 21:13:53] [INFO   ] Caught "cod" 
[2020-09-16 21:14:08] [INFO   ] Caught "fishing_rod" 
[2020-09-16 21:14:08] [INFO   ] -> UNBREAKING III 
[2020-09-16 21:14:24] [INFO   ] Caught "salmon" 
```
Java version:
```
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1~18.04-b09)
OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
```

**Comment by MrKinau** [comment]
Thank you for your bugreport.
This might happen, because of a book is gliching somehow with a high velocity. Another reason could be a high ping. Do you know if the book is glitching somehow (i.e. bouncing on stairs)? Or do you have a high ping to the server?
I'll set a better caught detection on my todo list (I think since 1.16 the item can be identified as loot from the fishing rod much easier)

**Comment by WantedFreak** [comment]
The latency to the server is shifting between 29ms and 35ms trough out the day.
```
20 packets transmitted, 20 received, 0% packet loss, time 19026ms
rtt min/avg/max/mdev = 29.646/29.757/29.853/0.216 ms
```
The bot is standing on 2x2 hoppers 7 blocks above sea-level so it could be glitching on them, but the one (only) time I went to the bot when it happened the book had been transported to a chest 10 blocks away with hoppers/dispenser and water.

**Comment by xantastic80** [comment]
[log0.log](https://github.com/MrKinau/FishingBot/files/5395613/log0.log)

**Comment by MrKinau** [comment]
This log does not seem to be related to this issue. Please open a new issue and describe your problem there @xantastic80

**Comment by MrKinau** [comment]
Over three years later, this should be fixed now, lol ([2.11.0](https://github.com/MrKinau/FishingBot/releases/tag/v2.11.0))
